File: backend/ec_customer/product/filters/product.py
from ec_base.common.constant.db_fields import CommonFields, ProductFields, MasterFields


class ProductListQueryFields:
    SEARCH_FIELDS = (
        ProductFields.NAME,
        "__".join(
            (ProductFields.CATEGORY, MasterFields.NAME),
        ),
    )
    ORDER_FIELDS = (
        CommonFields.ID,
        ProductFields.PRICE,
        ProductFields.QUANTITY,
    )
    ORDER_DEFAULT_FIELD = f"{CommonFields.ID}"
    FILTERSET_FIELDS = (ProductFields.CATEGORY_ID,)

File: frontend/ec_staff/src/interfaces/Product.ts
export interface Product {
  id: number | null,
  name: string,
  purchasePrice: number,
  price: number,
  desc: string,
  categoryId: number,
  photoId: null | number,
  quantity: null | number,
}

export interface ProductQuantity {
  id: number,
  quantity: number,
}
